contracts/contracts/levels/Motorbike.sol

```diff
@@ -41,6 +41,10 @@ contract Motorbike {
     fallback () external payable virtual {
         _delegate(_getAddressSlot(_IMPLEMENTATION_SLOT).value);
     }
+
+    function implementation() public returns(address) {
```

Comment:
It's not good to have a public function like this in the proxy (that's the origin of the transparent proxy pattern). It may be better to store a mapping from motorbike to engine in the factory storage.